e2e/test/TwinE2ETests.cs

```diff
@@ -246,10 +307,11 @@ public static async Task Twin_DeviceSetsReportedPropertyAndGetsItBackAsync(Devic
             await deviceClient.UpdateReportedPropertiesAsync(props).ConfigureAwait(false);
 
             Twin deviceTwin = await deviceClient.GetTwinAsync().ConfigureAwait(false);
-            Assert.AreEqual<string>(deviceTwin.Properties.Reported[propName].ToString(), propValue);
+            var actual = deviceTwin.Properties.Reported[propName];
+            Assert.AreEqual(JsonConvert.SerializeObject(actual), JsonConvert.SerializeObject(propValue));
         }
 
-        public static async Task<Task> SetTwinPropertyUpdateCallbackHandlerAsync(DeviceClient deviceClient, string expectedPropName, string expectedPropValue)
+        public static async Task<Task> SetTwinPropertyUpdateCallbackHandlerAsync(DeviceClient deviceClient, string expectedPropName, object expectedPropValue)
         {
```

Comment:
Does this mean if I used a string earlier to receive output "expectedPropValue", would it be broken now?

Reply:
No. The SDK always returned a map of <string, object>. It was just the tests that assumed the values would always be strings